test_semver.py

```diff
@@ -580,3 +584,44 @@ def test_should_be_able_to_use_integers_as_prerelease_build():
     assert isinstance(v.prerelease, str)
     assert isinstance(v.build, str)
     assert VersionInfo(1, 2, 3, 4, 5) == VersionInfo(1, 2, 3, '4', '5')
+
+
+@pytest.mark.parametrize("version,parts,expected", [
+    ("3.4.5", dict(major=2), '2.4.5'),
+    ("3.4.5", dict(major="2"), '2.4.5'),
+    ("3.4.5", dict(major=2, minor=5), '2.5.5'),
+    ("3.4.5", dict(minor=2), '3.2.5'),
+    ("3.4.5", dict(major=2, minor=5, patch=10), '2.5.10'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1"),
+     '2.5.10-rc1'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1", build="b1"),
+     '2.5.10-rc1+b1'),
+    ("3.4.5-alpha.1.2", dict(major=2), '2.4.5-alpha.1.2'),
+    ("3.4.5-alpha.1.2", dict(build="x1"), '3.4.5-alpha.1.2+x1'),
+    ("3.4.5+build1", dict(major=2), '2.4.5+build1'),
+])
+def test_should_replace(version, parts, expected):
+    assert replace(version, **parts) == expected
+
+
+def test_should_raise_typeerror_for_invalid_replace():
+    with pytest.raises(KeyError):
+        assert replace("1.2.3", unknown="should_raise")
+
+
+@pytest.mark.parametrize("version,parts,expected", [
+    ("3.4.5", dict(major=2, minor=5), '2.5.5'),
+    ("3.4.5", dict(major=2, minor=5, patch=10), '2.5.10'),
+    ("3.4.5-alpha.1.2", dict(major=2), '2.4.5-alpha.1.2'),
+    ("3.4.5-alpha.1.2", dict(build="x1"), '3.4.5-alpha.1.2+x1'),
+    ("3.4.5+build1", dict(major=2), '2.4.5+build1'),
+])
+def test_should_versioninfo_replace(version, parts, expected):
```

Comment:
Not sure what others think, but I'd fuse the 2 test cases to reuse the parameters.   
This makes it clear that `replace(version, **parts) ==  VersionInfo.parse(version).replace(**parts) == expected` (modulo string conversion)